runtime/vm/ClassInitialization.cpp

```diff
@@ -62,6 +62,33 @@ static char const *statusNames[] = {
 static j9object_t setInitStatus(J9VMThread *currentThread, J9Class *clazz, UDATA status, j9object_t initializationLock);
 static void classInitStateMachine(J9VMThread *currentThread, J9Class *clazz, J9ClassInitState desiredState);
 
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+static BOOLEAN
+compareRAMClasses(void *item, J9StackElement *currentElement)
+{
+	BOOLEAN rc = FALSE;
+
+	if ((J9Class *)currentElement->element == (J9Class *)item) {
```

Comment:
If the types here are both `void*`, there's no need for the casting. Function could be a single line (if `TRUE` and `FALSE` aren't identical to the C comparison results, we're in trouble).